src/lib.rs

```diff
@@ -234,6 +238,16 @@ impl Url {
         unsafe { ffi::ada_get_href(self.0) }.as_str()
     }
 
+    /// Updates the href of the URL, and triggers the URL parser.
+    /// Returns true if operation is successful.
```

Comment:
@anonrig It can fail?

Reply:
Yes, the URL specification contains certain edge cases.

Reply:
Then I think it is more idiomatic to return a `Result<(), UrlError>` instead. `UrlError` could be `()` if there is only one error variant.

For reference, the `url` crate provides various setter methods: https://docs.rs/url/2.4.1/url/struct.Url.html
(none of them are named `set_href` though)

Reply:
Makes sense. Would you like to open a pull request?